File: study-party-fe/src/services/auth.service.ts
import http from "@/lib/http.ts";
import type { AuthResponse, LoginPayload, RegisterPayload } from "@/types/auth.type.ts";
import { clearTokens } from "@/lib/token.ts";
import type {TokenPair} from "@/types/token.type.ts";

export const authService = {
	login: (payload: LoginPayload) => {
		return http.post<AuthResponse>("auth/login", payload, { withCredentials: true });
	},

	register: (payload: RegisterPayload) => {
		return http.post<void>("auth/register", payload,  { withCredentials: true });
	},

	logout: async () => {
		try {
			await http.post("/auth/logout");
		} catch (error) {
			console.error("Lỗi logout API:", error);
		} finally {
			clearTokens();
		}
	},

	loadMe: () => {
		return http.get("user/me");
	},

	refreshToken: () => {
		return http.post<TokenPair>("auth/refresh", {}, { withCredentials: true });
	},
}

File: party/src/main/java/com/web/services/auth/AuthService.java
package com.web.services.auth;

import com.web.dto.TokenPair;
import com.web.dto.request.user.LoginRequest;
import com.web.dto.request.user.RegisterRequest;
import com.web.study.party.dto.request.user.*;
import com.web.dto.response.TokenResponse;
import com.web.dto.response.auth.AuthResponse;

public interface AuthService {
    void register(RegisterRequest req);
    AuthResponse login(LoginRequest req, String ip, String ua);
    TokenResponse googleLogin(String idToken);
    TokenPair refresh(String refreshToken, String ip, String ua);
    void logout(String refreshToken);
}
